crates/uv-resolver/src/lock.rs

```diff
@@ -829,12 +827,15 @@ impl std::fmt::Display for DistributionId {
     }
 }
 
+/// A unique identifier to differentiate between different distributions for the same version of a
+/// package.
```

Comment:
I don't know that ID is right here, I defer to @BurntSushi.

Reply:
Yeah I'm not sure that `SourceId` is right here either. If it's an ID, then what is it identifying? For `DistributionId`, it's identifying a `Distribution`. A `Source` isn't really identifying anything, it is the source itself.

Reply:
I'm looking for different because `Source` makes it sound like we're reading the origin of a distribution from it, but (if i'm reading the code correctly) this field merely exists to match distributions with different versions for the same package in `DistributionId`, it's more like a subdivider.

Reply:
Should we just leave it as `Source` right now so that we can merge without resolving?

Reply:
Done